Tutorial: Rewrite React cookie banner (#8943)
PostHog/posthog.com: contents/tutorials/react-cookie-banner.md

title: Building a tracking cookies consent banner in React
sidebar: Docs
showTitle: true
featuredTutorial: false
date: 2024-06-10
author:
  - ian-vanagas
tags:
  - configuration
  - product os
---

> If you’ve spent any time online, you’ve seen a cookie consent banner. Because of GDPR and other worldwide internet privacy regulations, some sites need to get consent to track users and use cookies. Along with this is the need to provide, visitors with an easy way to opt in or out.

Reviewer: Making a bit shorter 
```suggestion
If you’ve spent any time online, you’ve seen a cookie consent banner. Because of GDPR and other worldwide internet privacy regulations, most sites need to get consent to track users and use cookies. 
```

> PostHog uses first-party cookies to identify and track users as well as manage user and application information across sessions. They enable us to more accurately track users and help you build better experiences for your users.

Reviewer: Do we need this paragraph? Intro is a bit long
```suggestion
```

To help you stay compliant, PostHog enables you to track users either with or without cookies. In this tutorial, we are going to build a simple banner with React, JavaScript, and HTML for users to opt in or out of PostHog’s cookies.

To complete this tutorial, you’ll need to a [PostHog instance](https://us.posthog.com/signup) as well as [Node installed](https://nodejs.org/en/download/) (to install, manage, and run React).

## Step 1: Setting up React

> This tutorial is for React, but this works for other JavaScript frameworks as well (we have specific tutorials for [Next.js](/tutorials/nextjs-cookie-banner) and [Vue](/tutorials/vue-cookie-banner)).

Reviewer: ```suggestion
This tutorial is for React, but it works for other JavaScript frameworks as well (we have specific tutorials for [Next.js](/tutorials/nextjs-cookie-banner) and [Vue](/tutorials/vue-cookie-banner)).
```

npm install --save posthog-js
```

> Next, we’ll go to `src/main.jsx` to get PostHog initialized. We’ll import PostHog and the `PostHogProvider` component from `posthog-js/react` and run `posthog.init` with our project key and instance address (which you can get in your project settings)

Reviewer: ```suggestion
Next, we’ll go to `src/main.jsx` to initialize PostHog. We’ll import PostHog and the `PostHogProvider` component from `posthog-js/react` and run `posthog.init` with our project key and instance address (which you can get in your [project settings](needs link))
```

```js
import React from 'react'
import ReactDOM from 'react-dom/client'
import App from './App.jsx'
import './index.css'
import posthog from 'posthog-js'
import { PostHogProvider } from 'posthog-js/react'

posthog.init("<ph_project_api_key>", {
  api_host: "<ph_client_api_host>",
})

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <PostHogProvider client={posthog}>
      <App />
    </PostHogProvider>
  </React.StrictMode>,
)

```

> Once done, visit your site running locally, click the button, and you start to see events in your PostHog instance. For now, the captured events are only the [autocaptured](/docs/data/autocapture) ones, but you can use custom events more.

Reviewer: ```suggestion
Once done, visit your site running locally, click the button, and you should start seeing events in your [PostHog activity tab](needs link). For now, the captured events are only the [autocaptured](/docs/data/autocapture) ones, but you can use custom events more.
```
